src/createServer.js

```diff
@@ -1,11 +1,190 @@
 'use strict';
 
 // const express = require('express');
+const express = require('express');
+
+const { Expenses } = require('./services/Expenses.js');
+const { Users } = require('./services/Users.js');
 
 function createServer() {
   // Use express to create a server
   // Add a routes to the server
   // Return the server (express app)
+  const app = express();
+
+  const expenses = new Expenses();
+  const users = new Users();
+
+  app.use(express.json());
+
+  app.get('/users', (req, res) => {
+    res.send(users.getUsers());
+  });
+
+  app.get('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+
+    const foundUser = users.getUserById(userId);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.send(foundUser);
+  });
+
+  app.post('/users', (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    res.statusCode = 201;
+    res.send(users.createUser(name));
+  });
+
+  app.delete('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+
+    const foundUser = users.getUserById(userId);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    users.deleteUser(userId);
+    res.sendStatus(204);
+  });
+
+  app.patch('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+    const { name } = req.body;
+
+    const foundUser = users.getUserById(userId);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    if (!name) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const updatedUser = users.updateUser(userId, name);
+
+    res.statusCode = 200;
+    res.send(updatedUser);
+  });
+
+  app.get('/expenses', (req, res) => {
+    const searchParams = req.query;
+
+    if (!searchParams) {
+      res.send(expenses.getExpenses());
+
+      return;
+    }
```

Comment:
I'm not sure, but `req.query` it's an object, so that condition cant be falsy, isn't?